docs/proposals/helm-charts.md

```diff
@@ -0,0 +1,206 @@
+    base_image: "docker.io/djzager/helm-bundle-base:latest"
+    white_list:
+      - ".*"
+```
+
+1. Type: refers to the registry adapter to handle this registry
+1. Name: gives a name to this registry item
+1. URL: the URL for the Helm chart repository
+1. Base Image: the container image to use when provisioning/deprovisioning a
+   Helm chart
+1. White List/Black List: allows the Admin to filter out Helm charts
+
+### Changes to the Registry Package
+
+- Add the ability to read the `base_image` from the broker config.
+- Pass the URL and BaseImage to the registry adapter.
+- Instantiate a Helm registry adapter to handle Helm Chart registries in the
+  config.
+
+### Creating a Helm Adapter
+
+The `HelmAdapter` struct will look something like the example below:
+
+```
+// import "k8s.io/helm/pkg/repo"
+// HelmAdapter - Helm Registry Adapter
+type HelmAdapter struct {
+    Config Configuration
+    Log    *logging.Logger
+    Charts map[string][]*repo.ChartVersion
+}
+```
+
+The addition of the `Charts` field makes it possible to save our work and only
+read the registries' index file once.
+
+When the broker calls `GetImageNames()` the Helm adapter will read the
+`index.yaml` found at the Helm Chart Repository URL based on the [chart
+repository
+structure](https://github.com/kubernetes/helm/blob/master/docs/chart_repository.md#the-chart-repository-structure).
+The broker will create an entry in the `Charts` field to hold a list of
+`ChartVersion` objects and  add the chart name to the list of `imageNames`
+to be returned.
+
+After these image names are filtered, the subsequent call to `FetchSpecs()`
+will:
+
+- Find the chart by name in the `Charts` field
+- Save all the versions of the chart:
+
+```
+var chartVersions []string
+for _, chart := range charts {
+    chartVersions = append(chartVersions, chart.Version)
+}
+```
+
+- Use the latest version of the chart to create a bundle `Spec` object:
+
+```
+// Use the latest chart for creating the bundle
+chart := charts[0]
+```
+
+- Load the chart in memory to grab the content of the values file:
+
+```
+resp, err := http.Get(chart.URLs[0])
+if err != nil {
+        return specs, err
+}
+defer resp.Body.Close()
+
+helmChart, err := chartutil.LoadArchive(resp.Body)
+if err != nil {
+        return specs, err
+}
+
+if helmChart.Values != nil {
+        values = helmChart.Values.Raw
+}
+```
+
+- Create a bundle `Spec` object:
+
+```
+// Convert chart to Bundle Spec
+spec := &apb.Spec{
+    Runtime:     2,
+    Version:     "1.0",
+    Async:       "optional",
+    Bindable:    false,
+    Image:       r.Config.BaseImage,
+    FQName:      chart.Name,
+    Tags:        chart.Keywords,
+    Description: chart.Description,
+    Metadata: map[string]interface{}{
+        //"longDescription":  chart.Description,
+        "displayName":      fmt.Sprintf("%s (Helm)", chart.Name),
+        "documentationUrl": chart.Home,
+        "dependencies":     chart.Sources,
+        "imageUrl":         chart.Icon,
```

Comment:
I still haven't been able to get this to work. Is there documentation somewhere that we can cite as the authority that this is the right key to use?

Reply:
This is strange. I have had no issues getting this to work.